### PR TITLE
Office-js: Adding Overloaded Fields to Item Subclasses

### PR DESCRIPTION
**Pull request by AlexJerabek**
AppointmentCompose, AppointmentRead, MessageCompose, and MessageRead need their full field list to surface in documentation. Pushing these overloads down also lets the documentation correctly specify individual modes.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://dev.office.com/reference/add-ins/outlook/preview/Office.context.mailbox.item
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by AlexJerabek** [comment]
Probably going to need @Zlatkovsky to check over this. If you want to do an in-person Code Review, that's totally fine.

File: types/office-js/index.d.ts [hunk 1]
```diff
@@ -7621,23 +7374,3155 @@ declare namespace Office {
...
+         * Gets an array of all the entities of the specified entity type found in the selected item.
+         *
+         * Note: This method is not supported in Outlook for iOS or Outlook for Android.
+         *
+         * [Api set: Mailbox 1.0]
+         *
+         * @returns
+         * If the value passed in entityType is not a valid member of the EntityType enumeration, the method returns null. If no entities of the specified type are present on the item, the method returns an empty array. Otherwise, the type of the objects in the returned array depends on the type of entity requested in the entityType parameter.
+         *
+         * While the minimum permission level to use this method is Restricted, some entity types require ReadItem to access, as specified in the following table.
```
Comment (Zlatkovsky) [inline types/office-js/index.d.ts]: How will a table look in regular IntelliSense usage (e.g., try it in TypeScript playground).  I'm guessing "badly".  If so, please move to remarks (here and elsewhere)

File: types/office-js/index.d.ts [hunk 2]
```diff
@@ -7621,23 +7374,3155 @@ declare namespace Office {
...
+         * @param callback When the method completes, the function passed in the callback parameter is called with a single parameter of type asyncResult. On success, the attachment identifier will be provided in the asyncResult.value property. If uploading the attachment fails, the asyncResult object will contain an Error object that provides a description of the error.
+         */
+        addFileAttachmentAsync(uri: string, attachmentName: string, callback: (result: AsyncResult) => void): void;
+
+        /**
+         * Adds an Exchange item, such as a message, as an attachment to the message or appointment.
+         *
+         * The addItemAttachmentAsync method attaches the item with the specified Exchange identifier to the item in the compose form. If you specify a callback method, the method is called with one parameter, asyncResult, which contains either the attachment identifier or a code that indicates any error that occurred while attaching the item. You can use the options parameter to pass state information to the callback method, if needed.
+         *
+         * You can subsequently use the identifier wi
```
Comment (Zlatkovsky) [inline types/office-js/index.d.ts]: Interesting.  Outlook is including Preview APIs?
If so, they should at least be marked with "@beta", which I believe the docs generation will also mark elsehow in the docs.

**Review by Zlatkovsky — APPROVED** [review]

File: types/office-js/index.d.ts (hunk 1 above)
Comment (AlexJerabek) [reply types/office-js/index.d.ts]: Good catch. I'll pull them into remarks.

File: types/office-js/index.d.ts (hunk 2 above)
Comment (AlexJerabek) [reply types/office-js/index.d.ts]: I'll make a separate work item to tag everything as beta.

**Comment by sandersn** [comment]
Unfortunately, it looks like the other Office-js PR conflicts with this one. @AlexJerabek Can you fix the conflicts please?

**Comment by AlexJerabek** [comment]
@sandersn Thanks for the heads up. Should be all ready to go now. Do I need owner approval again?

**Comment by sandersn** [comment]
Nope, I think you wrote all 3 of the recent PRs anyway. :)

**Pull request merged**